dds_web/api/errors.py

```diff
@@ -238,9 +254,15 @@ def __init__(self, message="No method found in request."):
 class KeyNotFoundError(exceptions.HTTPException):
     """Key not found in database."""
 
-    def __init__(self, project, message="No key found for current project"):
+    def __init__(self, project, message="No key found for current project", pass_message=False):
         self.message = f"{message}: {project}"
-        super().__init__(self.message)
+
+        if pass_message:
+            super().__init__(self.message)
+        else:
+            super().__init__("Unrecoverable encryption error. Aborting.")
+
+        general_logger.warning(self.message)
```

Comment:
Also, the different exceptions are basically using the same code for checking the pass_message so I'm wondering if we should have a base custom exception with that code in there, and then the rest inheriting from that class?

Reply:
I only introduced this parameter for selected errors, where I felt it would be specifically inappropriate to expose the details of the error to the users. So the majority of the errors doesn't need that code. 

I am not strictly opposed to introducing a second superclass to handle that part, but in relation to writing this simple if-statement out for a few times, it didn't really seem worth the effort to me. I guess there is little gain in terms of preventable code duplication compared to the complexity of juggling between two superclasses the errors will then inherit from. Unless we decide that all errors should have that parameter, I would say it is unnecessary perfectionism at this point?

Reply:
Agreed. Skip this for now 👍🏻